main/eval/src/mill/eval/Evaluator.scala

```diff
@@ -21,7 +21,14 @@ trait Evaluator {
   def outPath: os.Path
   def externalOutPath: os.Path
   def pathsResolver: EvaluatorPathsResolver
+  // TODO In 0.13.0, workerCache should have the type of mutableWorkerCache,
+  // while the latter should be removed
   def workerCache: collection.Map[Segments, (Int, Val)]
```

Comment:
Althouh this must stay for compat, we already can deprecate it in favor of its replacement. We try to deprecate API we intend to remove, to provide a smoother migration path.

Reply:
I was imagining a simple change to `def workerCache: collection.mutable.Map[Segments, (Int, Val)]` in `0.13`, and a removal of `mutableWorkerCache`, that should only be used internally. It's a really small change.

Are there know users of the `Evaluator` trait, that don't rely on the Mill internal implementation? Implementing that seems more of an internal thing to me… (and for users of `Evaluator`, given that `collection.mutable.Map <: collection.Map`, the change should be source-compatible).

Reply:
For now the policy is that the things that are exposed we treat as stable. We don't actually know if anyone uses it, but if we expect something to be internal-only we'll mark them as `private[mill]` and MIMA will let us do as we like

Reply:
The current state of the PR doesn't break bin compat. But the question is, should this (minor and mostly source-compatible) change go through a deprecation cycle, or should we do it in one go in `0.13`? I'd be ok with a change in one go, for this particular minor change